src/leap/mx/mail_receiver.py

```diff
@@ -378,10 +385,32 @@ def _step_process_mail_backend(self, filepath):
                 defer.returnValue(None)
 
             log.msg("Encrypting message to %s's pubkey" % (uuid,))
-            doc = yield self._encrypt_message(pubkey, mail_data)
+            try:
+                doc = yield self._encrypt_message(pubkey, mail_data)
+
+                yield self._export_message(uuid, doc)
+                yield self._remove(filepath)
+            except Exception as e:
+                yield self._bounce_with_timeout(filepath, msg, e)
+
+    @defer.inlineCallbacks
+    def _bounce_with_timeout(self, filepath, msg, error):
+        if filepath not in self._bounce_timeout:
+            self._bounce_timeout[filepath] = datetime.now()
```

Comment:
I find this naming scheme confusing. If you're storing a datetime object, better call it a timestamp, instead of a timeout? timeout suggest a delta.